server/bindings.go

```diff
@@ -6,6 +6,7 @@ import "github.com/streadway/amqp"
 // declared and bound. If the ExchangeName is not defined (an empty string), the
 // queue will not be bound to the exchange but assumed to use the default match.
 type HandlerBinding struct {
+	QueueName    string
```

Comment:
Not sure I get why this is needed, we use the provided name already so this must be that a topic binding **must** have a different queue name when declaring the queue and another when binding it? Was there any collisions?

Because this is what we did before:

```go
queue, err := inputCh.QueueDeclare(         
    binding.RoutingKey,  // This is what was sent to `TopicBinding` and still is                    
    s.queueDeclareSettings.Durable,         
    s.queueDeclareSettings.DeleteWhenUnused,
    s.queueDeclareSettings.Exclusive,       
    s.queueDeclareSettings.NoWait,          
    s.queueDeclareSettings.Args,            
)                                           
// ...
err = inputCh.QueueBind(          
    queue.Name, // This is the queue name which will be generated (previously proivided)                  
    binding.RoutingKey,  // And this is the same again.         
    binding.ExchangeName,         
    s.queueDeclareSettings.NoWait,
    binding.BindHeaders,          
)                                 
```

Reply:
I had to rewrite the Topic exchange test because of the graceful shutdown got stuck when the `chan` didn't have space in the handler. I'll revert the changes to `HandlerBinding` as this has nothing to do with this PR.

But some thoughts on this:

Having the queue name auto-generated will make each each server started (multiple instances) receive the same message. This could sometimes be what we want, but **definitely** not always.

Having the queue name the same as the routing key will have the message delivered only once for that routing key, no matter how many subscribers (could be different applications). This could sometimes be what we want, but like the other option **definitely** not always.